OSDOCS-20040-2: Adds PlanPolicy to RHCL (#114514)
openshift/openshift-docs: modules/proc-rhcl-planpolicy-verify.adoc

// Module included in the following assemblies:
//
// *deployment/rhcl-planpolicy.adoc

:_mod-docs-content-type: PROCEDURE
[id="proc-rhcl-planpolicy-verify_{context}"]

> = Verifying the acceptance of the 'AuthPolicy' CR

Reviewer: ```suggestion
= Verifying the acceptance of the AuthPolicy CR
```

[role="_abstract"]
You can confirm the acceptance of the `AuthPolicy` custom resource (CR) by running a few commands.

.Procedure

. Confirm that the `AuthPolicy` CR was accepted by checking its status by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ oc get authpolicy _<sample_app.yaml>_ -n ${KUADRANT_GATEWAY_NS} -o jsonpath='{.status.conditions[?(@.type=="Accepted")].status}'
----

> * Replace `_<sample_app.yaml>_` with the name of your application.

Reviewer: ```suggestion
+
* Replace `_<sample_app.yaml>_` with the name of your application.
```

* Replace ${KUADRANT_GATEWAY_NS} with the namespace you used for this Connectivity Link deployment.
+
.Expected output
[source,terminal,subs="+quotes"]
----
True
----

> . Check that your `AuthPolicy` has `Accepted` and `Enforced` status by running the following command:

Reviewer: ```suggestion

. Check that your `AuthPolicy` has `Accepted` and `Enforced` status by running the following command:
```

+
[source,terminal,subs="+quotes"]
----
$ oc get authpolicy ${KUADRANT_GATEWAY_NAME}-auth -n ${KUADRANT_GATEWAY_NS} -o=jsonpath='{.status.conditions[?(@.type=="Accepted")].message}{"\n"}{.status.conditions[?(@.type=="Enforced")].message}'
----

> * Replace 'KUADRANT_GATEWAY_NAME' with the name for your gateway in {product-title}.

Reviewer: ```suggestion
+
* Replace `KUADRANT_GATEWAY_NAME` with the name for your gateway in {product-title}.
```

> * Replace 'KUADRANT_GATEWAY_NS' with the namespace for your gateway in {product-title}.c

Reviewer: ```suggestion
* Replace `KUADRANT_GATEWAY_NS` with the namespace for your gateway in {product-title}.c
```

----
AuthPolicy has been accepted
AuthPolicy has been successfully enforced
----
+
. Confirm that all three API key secrets are present in the gateway namespace by running the following command:
+
[source,terminal,subs="+quotes"]
----
$ oc get secrets -n ${KUADRANT_GATEWAY_NS} -l app=_<sample_app.yaml>_
bronze-key   Opaque   1      2 mins
gold-key     Opaque   1      2 mins
----

> * Replace 'KUADRANT_GATEWAY_NS' with the namespace for your gateway in {product-title}.

Reviewer: ```suggestion
+
* Replace 'KUADRANT_GATEWAY_NS' with the namespace for your gateway in {product-title}.
```

* Replace `_<sample_app.yaml>_` with the name of your application.
+
. Send a request without an API key and confirm that it is rejected by running the following command:
+
[source,terminal,subs="+quotes"] 
----

> $ $ curl -H 'Authorization: APIKEY _<gold_api_key_value>_' http://_<your_api_hostname>_/toy -i

Reviewer: ```suggestion
$ curl -H 'Authorization: APIKEY _<gold_api_key_value>_' http://_<your_api_hostname>_/toy -i
```
